### PR TITLE
Highlight color picker

### LINKED ISSUE #61: Make visual highlighter change color
# Description
Add a color picker to settings which allows the user to change the color of the highlight. This way, the user can use any color they like. Improves accessibility.

# Definition of Done
* [x]  Plugin settings has color picker
* [x]  Color picker determines color of highlight in gutter
* [x]  Color picker determines color of highlight in mini-editors

### PR DESCRIPTION
**Pull request by bolkhod**
# Description of changes made
Changed highlight of the color highlighter in the TestGenie settings. Color of Gutter and mini-editor background will change accordingly. Cannot change transparency (intended).

# Why is merge request needed
This is an accessibility setting.

# Other notes
Closes #61 

*Please write if you have anything to add*

# What is missing?
Nothing. Reviews might want to check if clean up was done correctly (merging development was difficult)

- [x] I have checked that I am merging into correct branch

### CODE REVIEW

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt [hunk 1]
```diff
@@ -48,6 +54,10 @@ class SettingsPluginComponent {
             .addComponent(showCoverageCheckBox, 10)
             .addComponent(JXTitledSeparator("Environment settings"), 15)
             .addLabeledComponent(JBLabel("Java 11 path:"), javaPathTextField, 10, false)
+            // Add accessibility options
+            .addComponent(accessibilitySeparator)
+            .addComponent(JBLabel("Choose color for visualisation highlight"))
```
Comment (martinmladenov) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt:59]: You should add an inset to those two components in order to keep them consistent with the rest of the UI - 15 for the separator and 10 for the label.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt [hunk 2]
```diff
@@ -62,8 +72,18 @@ class SettingsPluginComponent {
         val width = panel?.visibleRect?.width
         val height = panel?.visibleRect?.height
 
+        // Simplify colorPicker
+        colorPicker.removeChooserPanel(colorPicker.chooserPanels.component1())
+        colorPicker.removeChooserPanel(colorPicker.chooserPanels.component2())
+        colorPicker.removeChooserPanel(colorPicker.chooserPanels.component2())
+        colorPicker.removeChooserPanel(colorPicker.chooserPanels.component2())
```
Comment (martinmladenov) [inline src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt:79]: Is this the only way to remove those panels? Is there to initialize the color picker without those components in the first place?

**Review by martinmladenov — CHANGES_REQUESTED** [review]

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt (hunk 2 above)
Comment (bolkhod) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt:79]: From my research and experimentation, this is unfortunately the only method. This component is older than some dinosaurs.

Comment (martinmladenov) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt:79]: Ohh, I see.

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/TestGenieSettingsState.kt [hunk 3]
```diff
@@ -24,7 +24,10 @@ data class TestGenieSettingsState(
     var criterionMethod: Boolean = true,
     var criterionMethodNoException: Boolean = true,
     var criterionCBranch: Boolean = true,
-    var minimize: Boolean = true
+    var minimize: Boolean = true,
+    var colorRed: Int = 100,
+    var colorGreen: Int = 150,
+    var colorBlue: Int = 20
```
[the remark this answers is not in the document]
Comment (bolkhod) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/TestGenieSettingsState.kt:30]: Thanks, will do

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt (hunk 2 above)
Comment (kirilvasilev16) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt:79]: 😳

**Review by kirilvasilev16 — COMMENTED** [review]
Can you also change the line coverage visualisation to make use of the new settings colors?

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/CoverageVisualisationService.kt [hunk 4]
```diff
@@ -38,7 +39,10 @@ class CoverageVisualisationService(private val project: Project) {
         // Show in-line coverage only if enabled in settings
         val state = ApplicationManager.getApplication().getService(TestGenieSettingsService::class.java).state
         if (state.showCoverage) {
-            val color = Color(100, 150, 20)
+            val editor = FileEditorManager.getInstance(project).selectedTextEditor!!
```
[the remark this answers is not in the document]
Comment (bolkhod) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/CoverageVisualisationService.kt]: did that

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt (hunk 1 above)
Comment (bolkhod) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginComponent.kt]: did that

File: src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/TestGenieSettingsState.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (bolkhod) [reply src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/TestGenieSettingsState.kt]: did that

**Comment by bolkhod** [comment]
> Can you also change the line coverage visualisation to make use of the new settings colors?

That feature is now included

**Review by martinmladenov — APPROVED** [review]
Thanks for addressing the comments, LGTM!

**Pull request merged**